formula_auditor: do not recommend `uses_from_macos` when macOS-only

cho-m opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

At least impacts `libiconv`.

e.g. `simdutf`

Before, recommendation is to change to `uses_from_macos "libiconv"` which doesn't make sense given it `depends_on :macos`
```console
❯ brew audit --new-formula simdutf
simdutf:
  * New formulae in homebrew/core should not have a `bottle do` block
  * Dependency 'icu4c' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
  * Dependency 'libiconv' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
Error: 3 problems in 1 formula detected
```

After,
```console
❯ brew audit --new-formula simdutf
simdutf:
  * New formulae in homebrew/core should not have a `bottle do` block
  * Dependency 'icu4c' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
  * Dependency 'libiconv' is provided by macOS; please remove it.
Error: 3 problems in 1 formula detected
```

MikeMcQuaid commented:
> Before, recommendation is to change to `uses_from_macos "libiconv"` which doesn't make sense given it `depends_on :macos`

Personally, I think it still does make sense and ideally we'd have these populated as informational, even when they are not required.

MikeMcQuaid commented:
> it will result in bad style as `uses_from_macos "libiconv"` currently is just `depends_on :macos` (i.e. no-op on macOS while unsupported dependency on Linux)

Do you mean you perceive this to be bad style or `brew style` fails?

Personally, I think this style is desirable.

cho-m commented:
> Do you mean you perceive this to be bad style or `brew style` fails?
> 
> Personally, I think this style is desirable.

Perceived as bad style & aligned to how we modified formulae in `homebrew-core`.

As an example, during linuxbrew migration, we corrected all usage of `uses_from_macos "libiconv"`, which had prevented Linux bottling:
- https://github.com/Homebrew/homebrew-core/commit/7e4762bb1ec845ff308add73d60a97569aba59f5
- https://github.com/Homebrew/homebrew-core/commit/34d4b269946d73d656a799dda3ebc8ff83aa991e
- https://github.com/Homebrew/homebrew-core/commit/42ab0891f520a967cfcbe538d46ad8e5718c0896
- https://github.com/Homebrew/homebrew-core/commit/94470624ea44b927c6645513e6bc44f72c72e4ce
- https://github.com/Homebrew/homebrew-core/commit/f27ff70dc51a274becf9651d385adce46faa3857
- https://github.com/Homebrew/homebrew-core/commit/a3359b99246ebb0983d1b314c9b82ace725d5c18
- https://github.com/Homebrew/homebrew-core/commit/cb13c37428e6f2edf16818dbf719068c4c43b5e0
- https://github.com/Homebrew/homebrew-core/commit/3eeff3db54240970124f29fc872c151ab8f9514a

---

Currently, there is 0 usage of `uses_from_macos "libiconv"` or `depends_on "libiconv"` in `homebrew-core` and it should ideally remain this way.

I will admit though that `libiconv` is a bit of an outlier as macOS provides both the libraries & header while Linux includes it in glibc.

MikeMcQuaid commented:
> Currently, there is 0 usage of `uses_from_macos "libiconv"` or `depends_on "libiconv"` in `homebrew-core` and it should ideally remain this way.

Personally I think it'd be nicer to use it as metadata and fix how Homebrew/brew handles it to avoid preventing bottling.

MikeMcQuaid commented:
Closing this due to lack of consensus, sorry.
